src/config/Config.zig

```diff
@@ -2911,20 +2913,44 @@ fn writeConfigTemplate(path: []const u8) !void {
     );
 }
 
-/// Load configurations from the default configuration files. The default
-/// configuration file is at `$XDG_CONFIG_HOME/ghostty/config`.
+/// Load configurations from the default configuration files.
+/// The default system configuration files are at `$XDG_CONFIG_DIRS/ghostty/config`.
+/// The default user configuration file is at `$XDG_CONFIG_HOME/ghostty/config`.
 ///
 /// On macOS, `$HOME/Library/Application Support/$CFBundleIdentifier/config`
 /// is also loaded.
 pub fn loadDefaultFiles(self: *Config, alloc: Allocator) !void {
-    // Load XDG first
-    const xdg_path = try internal_os.xdg.config(alloc, .{ .subdir = "ghostty/config" });
+    const config_subdir = "ghostty";
+    const config_file = "config";
+    // Load XDG system config first
+    var it = internal_os.xdg.DirIterator.init(.config);
+    // We must reverse the order of the iterator because the first xdg config
+    // dir must takes importance per the spec https://specifications.freedesktop.org/basedir-spec/latest/#variables
+    var xdg_config_dirs = std.ArrayList([]const u8).init(alloc);
+    defer xdg_config_dirs.deinit();
```

Comment:
Instead of all this machinery, the `DirIterator` could do a reverse iteration using `std.mem.lastIndexOf` to implement your own iterator. I would recommend this route.